beatrix/src/main/java/org/killbill/billing/beatrix/extbus/BeatrixListener.java

```diff
@@ -293,31 +295,35 @@ private BusEvent computeExtBusEventEntryFromBusInternalEvent(final BusInternalEv
                 objectType = ObjectType.TAG;
                 objectId = realUserTagEventCr.getTagId();
                 eventBusType = ExtBusEventType.TAG_CREATION;
-                metaData = realUserTagEventCr.getTagDefinition().getName();
+                final TagMetadata userTagCreationMetadata = new TagMetadata(realUserTagEventCr.getTagId(), realUserTagEventCr.getObjectId(), realUserTagEventCr.getObjectType(), realUserTagEventCr.getTagDefinition());
+                metaData = objectMapper.writeValueAsString(userTagCreationMetadata);
                 break;
 
             case CONTROL_TAG_CREATION:
                 final ControlTagCreationInternalEvent realTagEventCr = (ControlTagCreationInternalEvent) event;
                 objectType = ObjectType.TAG;
                 objectId = realTagEventCr.getTagId();
                 eventBusType = ExtBusEventType.TAG_CREATION;
-                metaData = realTagEventCr.getTagDefinition().getName();
+                final TagMetadata controlTagCreationMetadata = new TagMetadata(realTagEventCr.getTagId(), realTagEventCr.getObjectId(), realTagEventCr.getObjectType(), realTagEventCr.getTagDefinition());
+                metaData = objectMapper.writeValueAsString(controlTagCreationMetadata);
                 break;
 
             case USER_TAG_DELETION:
                 final UserTagDeletionInternalEvent realUserTagEventDel = (UserTagDeletionInternalEvent) event;
                 objectType = ObjectType.TAG;
                 objectId = realUserTagEventDel.getTagId();
                 eventBusType = ExtBusEventType.TAG_DELETION;
-                metaData = realUserTagEventDel.getTagDefinition().getName();
+                final TagMetadata userTagDeletionMetadata = new TagMetadata(realUserTagEventDel.getTagId(), realUserTagEventDel.getObjectId(), realUserTagEventDel.getObjectType(), realUserTagEventDel.getTagDefinition());
+                metaData = objectMapper.writeValueAsString(userTagDeletionMetadata);
                 break;
 
             case CONTROL_TAG_DELETION:
                 final ControlTagDeletionInternalEvent realTagEventDel = (ControlTagDeletionInternalEvent) event;
                 objectType = ObjectType.TAG;
                 objectId = realTagEventDel.getTagId();
                 eventBusType = ExtBusEventType.TAG_DELETION;
-                metaData = realTagEventDel.getTagDefinition().getName();
+                final TagMetadata controlTagDeletionMetadata = new TagMetadata(realTagEventDel.getTagId(), realTagEventDel.getObjectId(), realTagEventDel.getObjectType(), realTagEventDel.getTagDefinition());
```

Comment:
Even if the `objectId` and `objectType` fields are included in the `TagMetadata` class, I do not think populating the `objectType` from the internal event would not be correct. So I would suggest the following:
```suggestion
final TagMetadata controlTagDeletionMetadata = new TagMetadata(realTagEventDel.getTagId(), realTagEventDel.getObjectId(), objectType, realTagEventDel.getTagDefinition());
```

Reply:
Hi @reshmabidikar 

I have changed the code as per the above comments